lib/binaries.sh

```diff
@@ -134,6 +134,36 @@ install_npm() {
   fi
 }
 
+install_yarn_using_corepack_package_manager() {
+  local package_manager="$1"
+  local node_version="$2"
+  install_corepack_package_manager "$package_manager" "$node_version"
+  suppress_output yarn --version
+  echo "Using yarn $(yarn --version)"
+}
+
+install_corepack_package_manager() {
```

Comment:
This doesn't cache the actual package manager anywhere. We'll download yarn and/or pnpm over the wire on each build. Corepack does have a cache of package manager data, which we could put in the build cache. That'd be a nice optimization potentially, though a different PR would probably make sense for that.

Reply:
Are we caching npm and Yarn already? From the code in [binaries.sh](https://github.com/heroku/heroku-buildpack-nodejs/blob/main/lib/binaries.sh) it doesn't appear to be the case...

Reply:
> Are we caching npm and Yarn already?

No, not in this builldpack, but we do in our CNB. It's definitely not mandatory, but a nice optimization. It might be worthwhile since there's an opportunity to cache both yarn and pnpm distributions with just one implementation (again, in another PR).

Reply:
👍 I'll poke around and see what I can get working for caching of the package manager.

Reply:
@joshwlewis this is implemented in https://github.com/heroku/heroku-buildpack-nodejs/pull/1232